### PR TITLE
feat(core,webserver): support conditions on webhooks

### LINKED ISSUE #2267: Webhook trigger : allow to specify condition on payload to trigger execution
### Feature description

For some APIs we might want to check a condition (key exists, value is a specific one we are interested in) to decide to trigger our flow or not.

Currently this is doable within the flow definition but add some complexity, still generate flow execution and pollute history.

=> Can be nice to have an optional parameter `payloadCondition` which would contain our condition.

Flow could look like : 

```YAML
triggers:
  - id: webhook
    type: io.kestra.core.models.triggers.types.Webhook
    key: 4wjtkzwVGBM9yKnjm3yv8r
    payloadCondition: {{ trigger.body.pull_request.comments_url ?? true : false }}
```

### PR DESCRIPTION
**Pull request by loicmathieu**
@anna-geller, I took the occasion to slightly improve the documentation, can you have a look?

close #2267

### CODE REVIEW

File: core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java [hunk 1]
```diff
@@ -31,12 +31,19 @@
...
+        Kestra accept `GET`, `POST` & `PUT` request on this url.
+
+        The whole body & headers will be available as variable:
+        - `trigger.body`
+        - `trigger.headers`
+
+        The webhook response can be:
+        - A status code 404 if the namespace, flow or webhook key is not found
+        - A status code 200 if the webhook trigger an execution
+        - A status code 204 if the webhook didn't trigger an execution due to webhook conditions that didn't match"""
```
Comment (anna-geller) [inline core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java:46]:
```suggestion
        Webhook trigger allows you to create a unique URL that you can use to trigger a Kestra flow execution based on a presence of events in another application such as GitHub or Amazon EventBridge. In order to use that URL, you have to add a secret key that will secure your webhook URL. 
        
        The URL will then follow the following format: `https://{your_hostname}/api/v1/executions/webhook/{namespace}/{flowId}/{key}`. Replace the templated values accordingly to your workflow setup.
        
        The webhook URL accepts `GET`, `POST` and `PUT` requests.

        You can access the request body and headers sent by another application using the following template variables:
        - `{{ trigger.body }}`
        - `{{ trigger.headers }}`.

        The webhook response will be one of the following HTTP status codes:
        - 404 if the namespace, flow or webhook key is not found
        - 200 if the webhook triggers an execution
        - 204 if the webhook cannot trigger an execution due to a lack of matching event conditions sent by other application."""
```

File: core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java [hunk 2]
```diff
@@ -49,6 +56,21 @@
                 "    key: 4wjtkzwVGBM9yKnjm3yv8r"
             },
             full = true
+        ),
+        @Example(
+            title = """
+                Add a webhook trigger with a condition, the flow will be executed only if the condition met.`.
```
Comment (anna-geller) [inline core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java:62]:
```suggestion
                Add a trigger matching specific webhook event condition. The flow will be executed only if the condition is met.`.
```

File: core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java [hunk 3]
```diff
@@ -60,7 +82,7 @@ public class Webhook extends AbstractTrigger implements TriggerOutput<Webhook.Ou
         description = "The key is used for generating the url of the webhook.\n" +
             "\n" +
             "::alert{type=\"warning\"}\n" +
-            "Take care when using manual key, the key is the only security to protect your webhook and must be considered as a secret !\n" +
+            "Take care when choosing the key, it is the only security to protect your webhook and must be considered as a secret! We advise to use a random key generator to create the key.\n" +
```
Comment (anna-geller) [inline core/src/main/java/io/kestra/core/models/triggers/types/Webhook.java:85]:
```suggestion
            "Make sure to keep the webhook key secure. It's the only security mechanism to protect your endpoint from bad actors, and must be considered as a secret. You can use a random key generator to create the key.\n" +
```

File: core/src/test/resources/flows/valids/webhook-with-condition.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+id: webhook-with-condition
+namespace: io.kestra.tests
```
Comment (anna-geller) [inline core/src/test/resources/flows/valids/webhook-with-condition.yaml:2]:
```suggestion
namespace: dev
```

File: core/src/test/resources/flows/valids/webhook-with-condition.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+tasks:
+  - id: out
+    type: io.kestra.core.tasks.debugs.Return
+    format: "{{trigger | json }}"
```
Comment (anna-geller) [inline core/src/test/resources/flows/valids/webhook-with-condition.yaml:7]:
```suggestion
    format: "{{ trigger | json }}"
```
nitpick

**Review by anna-geller — APPROVED** [review]

File: core/src/test/resources/flows/valids/webhook-with-condition.yaml (hunk 4 above)
Comment (loicmathieu) [reply core/src/test/resources/flows/valids/webhook-with-condition.yaml:2]: We always use this namesapce in our tests, this is not something visible to users

Comment (loicmathieu) [reply core/src/test/resources/flows/valids/webhook-with-condition.yaml:2]: We always use this namesapce in our tests, this is not something visible to users

**Review by Skraye — APPROVED** [review]
LGTM

**Pull request merged**